### PR TITLE
Time: Remove serverside getter, and use atomic operations

### PR DESCRIPTION
**Pull request by est31**
It isn't possible to use atomic operators for floats, so don't use them there.

Having a lock is good out of other reasons too, because this way the float time
and the integer time both match, and can't get different values in a race,
e.g. when setTimeofDay() gets executed simultaneously.

PR also includes commit to clean up atomic.h, which was necessary because android build failed.

### CODE REVIEW

**Comment by est31** [comment]
* bug label because we actually do a lock
* enhancement label because now we use atomics which is an enhancement
* maintenance label because of the serverside getter remove

**Comment by nerzhul** [comment]
:+1:

File: src/threading/atomic.h [hunk 1]
```diff
@@ -29,64 +29,83 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
+			m_mutex.lock();            \
+			T _val = (op);             \
+			m_mutex.unlock();          \
+			return _val;               \
 		} while (0)
-	#define ATOMIC_LOAD(T, v) \
+	#define ATOMIC_LOCK_CAS(T, v, e, d) do { \
+			m_mutex.lock();                  \
+			bool _eq = (v == e);             \
+			if (eq)                          \
```
Comment (ShadowNinja) [inline src/threading/atomic.h]: `_eq` / `eq`

File: src/threading/atomic.h [hunk 2]
```diff
@@ -29,64 +29,83 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
+				v = d;                       \
+			m_mutex.unlock();                \
+			return eq;                       \
+		} while (0)
+	#define ATOMIC_LOAD(T, v)                     \
 		if (sizeof(T) <= sizeof(void*)) return v; \
 		else ATOMIC_LOCK_OP(T, v);
-	#define ATOMIC_STORE(T, v, x) \
+	#define ATOMIC_STORE(T, v, x)                     \
```
Comment (ShadowNinja) [inline src/threading/atomic.h]: This doesn't align the backslash with anything.

File: src/threading/atomic.h (hunk 1 above)
Comment (est31) [inline src/threading/atomic.h]: Ah thanks

File: src/threading/atomic.h (hunk 2 above)
Comment (est31) [inline src/threading/atomic.h]: It does if looked at with a tab width of 4: https://github.com/minetest/minetest/pull/3333/files?ts=4

Comment (ShadowNinja) [inline src/threading/atomic.h]: But our "official" tab width is 8.

Comment (est31) [inline src/threading/atomic.h]: We override it in the line limit rule already to be 4, and @kwolekr aligns his backslashes with a 4 space indent as well. I'm trying to get at least a bit of consistency in the codebase.

**Comment by est31** [comment]
abc354a5d0a941cf1f3877e487cce325cc609ba8
f9b09368f063cdace93a042d5bdd45987c084d94

**Pull request closed without merging**